superset-frontend/packages/superset-ui-chart-controls/src/components/ColumnOption.tsx

```diff
@@ -48,14 +52,27 @@ export function ColumnOption({
   const hasExpression = expression && expression !== column_name;
   const type = hasExpression ? 'expression' : type_generic;
   const [tooltipText, setTooltipText] = useState<ReactNode>(column.column_name);
+  const [columnTypeTooltipText, setcolumnTypeTooltipText] = useState<ReactNode>(
+    column.type,
+  );
 
   useLayoutEffect(() => {
     setTooltipText(getColumnTooltipNode(column, labelRef));
+    setcolumnTypeTooltipText(getColumnTypeTooltipNode(column));
   }, [labelRef, column]);
 
   return (
     <StyleOverrides>
-      {showType && type !== undefined && <ColumnTypeLabel type={type} />}
+      {showType && (
+        <Tooltip
+          id="metric-type-tooltip"
+          title={columnTypeTooltipText}
+          placement="bottomRight"
+          align={{ offset: [8, -2] }}
+        >
+          <span>{type !== undefined && <ColumnTypeLabel type={type} />}</span>
```

Comment:
We might not need this `type !== undefined` check if it gets added above.

Reply:
Removed this check from here in recent commit.